python/lsst/meas/astrom/pessimistic_pattern_matcher_b_3D.py

```diff
@@ -822,13 +823,17 @@ def _create_pattern_spokes(self, src_ctr, src_delta_array, src_dist_array,
 
             # Append the successful indices to our list. The src_idx needs
             # an extra iteration to skip the first and second source objects.
-            output_spokes.ref_spoke_list.append(ref_id)
-            output_spokes.src_spoke_list.append(src_idx + 1)
+            ref_spoke_list.append(ref_id)
+            src_spoke_list.append(src_idx + 1)
             # If we found enough reference objects we can return early. This is
             # n_match - 2 as we already have 2 source objects matched into the
             # reference data.
-            if len(output_spokes.ref_spoke_list) >= n_match - 2:
-                break
+            if len(ref_spoke_list) >= n_match - 2:
+                # Set the struct data and return the struct.
+                output_spokes.ref_spoke_list = ref_spoke_list
+                output_spokes.src_spoke_list = src_spoke_list
+                return output_spokes
+
         return output_spokes
 
     # TODO: Possibly clean up the arguments here by pre computing
```

Comment:
I realize this is pre-existing code but...can you please file a ticket for this cleanup and reference the ticket in the TODO code comment? That's how we are handling TODOs these days in order to keep track of them.